nvidia_triton/datadog_checks/nvidia_triton/check.py

```diff
@@ -0,0 +1,95 @@
+
+        # Get the API server port if specified, otherwise use the default 8000.
+        self.server_port = str(self.instance.get('server_port', 8000))
+
+        self.collect_server_info = self.instance.get('collect_server_info', True)
+
+        # Get the base url from the openmetrics endpoint and construct the server info API endpoint.
+        if self.collect_server_info:
+            parts = urlparse(self.openmetrics_endpoint)
+            # Delete the /metrics from the url
+            self.base_url = parts._replace(path="")
+            # Replace the openmetrics port by the server port
+            self.server_info_api = self.base_url._replace(netloc=parts.hostname + ':' + self.server_port).geturl()
+        else:
+            self.log.debug("Collecting server info through API is disabled.")
+            return
+
+    def get_default_config(self):
+        return {
+            "metrics": [METRICS_MAP],
+            # Rename labels that are reserved in datadog.
+            "rename_labels": {
+                "version": "triton_model_version",
+            },
+        }
+
+    @AgentCheck.metadata_entrypoint
+    def _submit_version_metadata(self):
+        endpoint = urljoin(self.server_info_api, self._metadata_endpoint)
+        response = self.http.get(endpoint)
+
+        if response.ok:
+            data = response.json()
+            version = data.get("version", "")
+            version_split = version.split(".")
+            if len(version_split) >= 3:
+                major = version_split[0]
+                minor = version_split[1]
+                patch = version_split[2]
+
+                version_raw = f'{major}.{minor}.{patch}'
+
+                version_parts = {
+                    'major': major,
+                    'minor': minor,
+                    'patch': patch,
+                }
+                self.set_metadata('version', version_raw, scheme='semver', part_map=version_parts)
+            else:
+                self.log.debug("Malformed Triton Server version format: %s", version)
+        else:
+            self.log.debug("Could not retrieve version metadata.")
+
+    def _check_server_health(self):
+        endpoint = urljoin(self.server_info_api, self._health_endpoint)
+        response = self.http.get(endpoint)
+
+        # Any 4xx or 5xx response from the API endpoint (/v2/health/ready) means the server is not ready
+        if 400 <= response.status_code and response.status_code < 600:
+            self.service_check('health.status', AgentCheck.CRITICAL, self.tags)
+        if response.status_code == 200:
+            self.service_check('health.status', AgentCheck.OK, self.tags)
+        else:
+            self.service_check('health.status', AgentCheck.UNKNOWN, self.tags)
+
+    def check(self, instance):
+        super().check(instance)
+        if self.collect_server_info:
+            self._submit_version_metadata()
+            self._check_server_health()
```

Comment:
Seems like the OMV2 implementation can raise exceptions https://github.com/DataDog/integrations-core/blob/6ffe167aa7e57fac9a01493b069910df2693817a/datadog_checks_base/datadog_checks/base/checks/openmetrics/v2/base.py#L77-L78

In that case, the server info collection won't happen. Would it make sense to `try/except` the `super().check(instance)` part or collect the server info before?

Reply:
Good catch! @HadhemiDD what do you think, does it make sense to collect metadata/server health even if the OM logic fails?

If yes, I'd argue in favor of putting `super().check(instance)` at the end. If metadata/server health stuff crashes, I think it's fine that we crash and investigate.

I'm taking note for future. It would be more robust to support calling multiple scrapers independently in OM2 base class, even if some of these scrapers aren't talking to prometheus endpoints. That would allow us to degrade more gracefully if one of the endpoints is down but not the other. In this particular case I feel that's overengineering.

Reply:
The try/except solution would be better, the server info URL is extracted from the openmetrics endpoint, so if we face an issue with the scraping of openmetrics endpoint, it is better to exist and investigate as the collect server info will most likely fail. @iliakur  @FlorentClarret What do you think?